lib/fizzy/instantiate.hpp

```diff
@@ -21,8 +21,31 @@ namespace fizzy
 struct ExecutionResult;
 struct Instance;
 
+class ThreadContext
```

Comment:
Why do we go with the term thread?  Is this future proofing for multithreaded wasm?  Is some kind of locking planned, given the memory (to begin with) is still part of the instance and not the thread.

Reply:
It has nothing to do with multithreading. I is just execution thread context (in principle, you can create multiple thread context and start multiple call stacks in parallel). I understand the name may be confusing although WABT uses the same name for this concept. We can consider alternative names.

Reply:
I think it would be a good name, but given wasm multithreading this may be confusing. Would this be applicable during that? Would something like `ExecutionContext` be less confusing, or equally confusing as it is partial too?